staging/src/k8s.io/legacy-cloud-providers/azure/azure_client.go

```diff
@@ -113,6 +113,7 @@ type VirtualMachineScaleSetVMsClient interface {
 
 // RoutesClient defines needed functions for azure network.RoutesClient
 type RoutesClient interface {
+	Get(ctx context.Context, resourceGroupName string, routeTableName string, routeName string) (result network.Route, err error)
```

Comment:
PR needs rebase. And please rename the returned `err error` to `rerr *retry.Error`, which is introduced in #86555.

Reply:
if we use az.ListRoutes(), then RoutesClient.Get() is no longer needed. I prefer not to include useless code, so I will remove this Get().

Reply:
Yep, it won't be required with LIST.